src/common/plugin/fulltext/elasticsearch/ESAdapter.cpp

```diff
@@ -0,0 +1,280 @@
+    return Status::OK();
+  }
+  auto error = resp["error"];
+  if (error.isObject()) {
+    return Status::Error(folly::toJson(error));
+  }
+  return Status::Error(folly::toJson(resp));
+}
+
+StatusOr<bool> ESAdapter::isIndexExist(const std::string& name) {
+  auto result = randomClient().getIndex(name);
+  if (!result.ok()) {
+    return result.status();
+  }
+  auto resp = std::move(result).value();
+  if (resp.count(name)) {
+    return true;
+  }
+  if (resp.count("status") && resp["status"].getInt() == 404) {
+    return false;
+  }
+  if (resp.count("error")) {
+    return Status::Error(folly::toJson(resp["error"]));
+  }
+  return Status::Error(folly::toJson(resp));
+}
+
+Status ESAdapter::bulk(const ESBulk& bulk, bool refresh) {
+  std::vector<folly::dynamic> jsonArray;
+  for (auto& [docId, objs] : bulk.documents_) {
+    for (auto& obj : objs) {
+      jsonArray.push_back(obj);
+    }
+  }
+  auto result = randomClient().bulk(jsonArray, refresh);
+  if (!result.ok()) {
+    return result.status();
+  }
+  auto resp = std::move(result).value();
+  if (!resp.count("error")) {
+    return Status::OK();
+  }
+  auto error = resp["error"];
+  if (error.isObject()) {
+    return Status::Error(folly::toJson(error));
+  }
+  return Status::Error(folly::toJson(resp));
+}
+
+StatusOr<ESQueryResult> ESAdapter::prefix(const std::string& index, const std::string& pattern) {
+  folly::dynamic body = folly::dynamic::object();
+  body["query"] = folly::dynamic::object();
+  body["query"]["prefix"] = folly::dynamic::object();
+  body["query"]["prefix"]["text"] = pattern;
+  return ESAdapter::query(index, body);
+}
+
+StatusOr<ESQueryResult> ESAdapter::wildcard(const std::string& index, const std::string& pattern) {
+  folly::dynamic body = folly::dynamic::object();
+  body["query"] = folly::dynamic::object();
+  body["query"]["wildcard"] = folly::dynamic::object("text", pattern);
+  return ESAdapter::query(index, body);
+}
+
+StatusOr<ESQueryResult> ESAdapter::regexp(const std::string& index, const std::string& pattern) {
+  folly::dynamic body = folly::dynamic::object();
+  body["query"] = folly::dynamic::object();
+  body["query"]["regexp"] = folly::dynamic::object("text", pattern);
+  return ESAdapter::query(index, body);
+}
+
+StatusOr<ESQueryResult> ESAdapter::fuzzy(const std::string& index,
+                                         const std::string& pattern,
+                                         const std::string& fuzziness) {
+  folly::dynamic body = folly::dynamic::object();
+  body["query"] = folly::dynamic::object();
+  body["query"]["fuzzy"] = folly::dynamic::object();
+  body["query"]["fuzzy"]["text"] = folly::dynamic::object("value", pattern)("fuzziness", fuzziness);
+  return ESAdapter::query(index, body);
+}
+
+// StatusOr<ESQueryResult> ESAdapter::term(const std::string& index,const std::vector<std::string>&
+// words){
+//   folly::dynamic query = folly::dynamic::object("query",folly::dynam)
+// }
```

Comment:
Can this commented code be removed?

Reply:
This part of the code will be used, but we haven't fully figured out how to implement it. Uncomment and test them in the next PR